# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/selection/crashtests/selection-modify-line-from-contenteditable-to-textarea.html
# Commit: e90eb44c2da3
# Full Hash: e90eb44c2da3303c038c67b7ad28b1498cd5a98c
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2024-07-19 16:21:39
# Description:
#   Bug 1897255 - Make `nsIFrame::PeekOffsetForLine` won't cross editing host boundary unless the callers allows that r=emilio
#   
#   The problems in the testcase are, `nsIFrame::PeekOffsetForLine` returns a frame
#   for any content node outside the editing host and `nsIFrame::GetLastLeaf`
#   returns a native anonymous subtree node if the frame is for the native
# ==============================================================================

diff -r bd02e7030721 -r e90eb44c2da3 testing/web-platform/tests/selection/crashtests/selection-modify-line-from-contenteditable-to-textarea.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/selection/crashtests/selection-modify-line-from-contenteditable-to-textarea.html	Fri Jul 19 08:13:10 2024 +0000
@@ -0,0 +1,35 @@
+<!doctype html>
+<html>
+<meta charset="utf-8">
+<style>
+:first-child {
+  display: table-header-group;
+  margin-right: 10px;
+}
+</style>
+<script>
+"use strict";
+
+addEventListener("DOMContentLoaded", () => {
+  const editingHost = document.querySelector("span[contenteditable]");
+  getSelection().collapse(editingHost, 0);
+  editingHost.blur();
+});
+
+function onfocusout() {
+  const selection = getSelection();
+  const range = selection.getRangeAt(0);
+  selection.modify("move", "left", "line");
+  selection.addRange(range);
+  document.execCommand("hiliteColor", false, "#5343B4");
+}
+</script>
+<body>
+<canvas> width="7"></canvas>
+<textarea>AAA</textarea>
+<fieldset onfocusout="onfocusout()">
+<span contenteditable>
+</span>
+</fieldset>
+</body>
+</html>
